### ISSUE TITLE
lecture panorama à la racine.

### ISSUE THREAD

**Issue by Manuel-SEILHES** [JeremyHeleine/Photo-Sphere-Viewer#37]
Bonjour.

Super appli ! Par contre, je voudrais afficher des panoramas constitués d'une image 3D unique. ça fonctionne très bien lorsque je l'importe à partir de GOOGLE par exemple. Mais GOOGLE réduis la taille de l'image ce qui déteriore sa qualité. Si je veux importer une image panorama directement de GOOGLE, elle n'est plus au format xxxx.jpg.
Serait il possible d'importer une image comme dans le ficher "example" qui lit une adresse, et comme dans le fichier "example1" qui peut lire un fichier non taggé directement à sa racine, sans connexion, que le fichier soit directement déclaré dans le script du container ???

En résumé, afficher un panorama en local comme dans exemple 1 mais programmable camme dans example.

Je vous ai mis une page  web avec un code qui fonctionne.

Merci d'avance
[index.txt](https://github.com/JeremyHeleine/Photo-Sphere-Viewer/files/384183/index.txt)

**Comment by JeremyHeleine** [comment]
Bonjour,

Je ne comprends pas ce que vous voulez faire. D'ailleurs, dans le fichier `example.html`, le fichier n'est pas dynamique mais bien hard-codé, tout comme dans le lien cliquable de `example1.html`.

**Comment by Manuel-SEILHES** [comment]
Bonjour.

 

Je voudrais afficher une image en local, comme dans l’example1 mais sans le bouton, comme dans l’example. Ce qui éviterait d’être dépendant des sites CORS comme GOOGLE ou FLICKR et permettrait de lire les images quelle que soit l’endroit où elle se trouvent.

 

Cordialement

 

Manuel SEILHES

N° SIREN :404 234 346

31 impasse du tarin

83210 LA FARLEDE

 

06.51.07.71.88

www.kubic3d.com <http://www.kubic3d.com/> 



 

 

 

De : Jérémy Heleine [mailto:notifications@github.com] 
Envoyé : jeudi 28 juillet 2016 12:44
À : JeremyHeleine/Photo-Sphere-Viewer <Photo-Sphere-Viewer@noreply.github.com>
Cc : Manuel-SEILHES <manuelseilhes@hotmail.com>; Author <author@noreply.github.com>
Objet : Re: [JeremyHeleine/Photo-Sphere-Viewer] lecture panorama à la racine. (#37)

 

Bonjour,

Je ne comprends pas ce que vous voulez faire. D'ailleurs, dans le fichier example.html, le fichier n'est pas dynamique mais bien hard-codé, tout comme dans le lien cliquable de example1.html.

**Comment by JeremyHeleine** [comment]
Pour le coup je ne vois pas trop la difficulté, il suffit de faire comme dans `example1.js` : dans le code appelé lors du clic sur le lien, c'est un panorama local qui est ouvert, mais sans passer par un upload.

**Comment by Manuel-SEILHES** [comment]
Je suis d’accord, justement, serait-il possible de le faire comme example1 sans le bouton ? L’idée c’est de n pas permettre d’interactivité entre l’utilisateur et la page, juste l’affichage ?

 

 

Manuel SEILHES

N° SIREN :404 234 346

31 impasse du tarin

83210 LA FARLEDE

 

06.51.07.71.88

www.kubic3d.com <http://www.kubic3d.com/> 



 

 

De : Jérémy Heleine [mailto:notifications@github.com] 
Envoyé : jeudi 28 juillet 2016 13:57
À : JeremyHeleine/Photo-Sphere-Viewer <Photo-Sphere-Viewer@noreply.github.com>
Cc : Manuel-SEILHES <manuelseilhes@hotmail.com>; Author <author@noreply.github.com>
Objet : Re: [JeremyHeleine/Photo-Sphere-Viewer] lecture panorama à la racine. (#37)

 

Pour le coup je ne vois pas trop la difficulté, il suffit de faire comme dans example1.js : dans le code appelé lors du clic sur le lien, c'est un panorama local qui est ouvert, mais sans passer par un upload.

**Comment by JeremyHeleine** [comment]
C'est tout à fait possible : dans le code de `example.html`, au lieu d'indiquer une URL, il suffit d'indiquer le chemin vers un fichier local.

**Comment by Manuel-SEILHES** [comment]
Pouvez-vous me donner un exemple de lien en direct ? je ne parviens pas à trouver le code correct (à ma décharge, mon expertise porte sur la 3D, que je voudrais utiliser pour créer des panoramas 3D, pas le code !!!)

 

Merci d’avance

 

Manuel SEILHES

N° SIREN :404 234 346

31 impasse du tarin

83210 LA FARLEDE

 

06.51.07.71.88

www.kubic3d.com <http://www.kubic3d.com/> 



 

 

De : Jérémy Heleine [mailto:notifications@github.com] 
Envoyé : jeudi 28 juillet 2016 14:52
À : JeremyHeleine/Photo-Sphere-Viewer <Photo-Sphere-Viewer@noreply.github.com>
Cc : Manuel-SEILHES <manuelseilhes@hotmail.com>; Author <author@noreply.github.com>
Objet : Re: [JeremyHeleine/Photo-Sphere-Viewer] lecture panorama à la racine. (#37)

 

C'est tout à fait possible : dans le code de example.html, au lieu d'indiquer une URL, il suffit d'indiquer le chemin vers un fichier local.

**Comment by JeremyHeleine** [comment]
If I take `example.html` as a base, below is the code you should use to load a local panorama.

```js
var div = document.getElementById('container');
var PSV = new PhotoSphereViewer({
		panorama: 'path/to/your/panorama.jpg',
		container: div
});
```

**Comment by robert-hoffmann** [comment]
Salut,

Ca ne marche pas ici non plus (win10), ni test.html, ni exemple1.html, (ko dans Chrome, IE11/Edge)

`PhotoSphereViewer.js:334 XMLHttpRequest cannot load file:///D:/Downloads/Photo-Sphere-Viewer-master/examples/sun.jpg. Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, https, chrome-extension-resource.`

**Comment by JeremyHeleine** [comment]
Bonjour,

Sans serveur web, il faut désactiver la lecture des données XMP (`usexmpdata` à `false`). Normalement ça devrait fonctionner sans problème.

**Comment by robert-hoffmann** [comment]
Bonjour,

J'avais testé ca mais du coup on as l'erreur

`Access to Image at 'file:///D:/Downloads/Photo-Sphere-Viewer-master/examples/sun.jpg' from origin 'file://' has been blocked by CORS policy: Invalid response. Origin 'null' is therefore not allowed access.`

La seul bidouille j'ai trouve (vu que electron utilise Node) est de lire l'imgage avec FS, et ensuite l'injecter en base64 image:data

Mais suis ouvert a d'autres solutions ! 

Merci

**Comment by theovillar** [comment]
Bonjour, 

Quand j'injecte en base64 mon image dans le pano cela fonctionne sur ordinateur mais pas sur mon iphone5s. 
Mon iphone5s accepte uniquement quand je met le path directement de l'image sur le serveur. 

Y a t'il une solution à ce problème ? 

Merci, 

Théo

**Comment by meer-online** [comment]
@theovillar This is an old post but did you find a solution to base64 images?
